In [4]:
import pandas as pd

In [5]:
df = pd.read_csv("../data/Sample - Superstore.csv", encoding="latin1")

In [6]:
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136
1,2,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820
2,3,CA-2016-138688,6/12/2016,6/16/2016,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714
3,4,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310
4,5,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164


In [7]:
df.shape

(9994, 21)

In [8]:
df.columns

Index(['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode',
       'Customer ID', 'Customer Name', 'Segment', 'Country', 'City', 'State',
       'Postal Code', 'Region', 'Product ID', 'Category', 'Sub-Category',
       'Product Name', 'Sales', 'Quantity', 'Discount', 'Profit'],
      dtype='object')

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   object 
 2   Order Date     9994 non-null   object 
 3   Ship Date      9994 non-null   object 
 4   Ship Mode      9994 non-null   object 
 5   Customer ID    9994 non-null   object 
 6   Customer Name  9994 non-null   object 
 7   Segment        9994 non-null   object 
 8   Country        9994 non-null   object 
 9   City           9994 non-null   object 
 10  State          9994 non-null   object 
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   object 
 13  Product ID     9994 non-null   object 
 14  Category       9994 non-null   object 
 15  Sub-Category   9994 non-null   object 
 16  Product Name   9994 non-null   object 
 17  Sales          9994 non-null   float64
 18  Quantity

In [10]:
df.isnull().sum()

Row ID           0
Order ID         0
Order Date       0
Ship Date        0
Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Postal Code      0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
Quantity         0
Discount         0
Profit           0
dtype: int64

In [11]:
df['Order ID'].nunique()

5009

In [12]:
df['Product ID'].nunique()

1862

In [13]:
df['Customer ID'].nunique()

793

In [14]:
df['Row ID'].nunique()

9994

In [15]:
df['Row ID'].duplicated().sum()

np.int64(0)

In [16]:
df['Order Date'] = pd.to_datetime(df['Order Date'])
df['Ship Date'] = pd.to_datetime(df['Ship Date'])

In [17]:
df[['Order Date', 'Ship Date']].dtypes

Order Date    datetime64[ns]
Ship Date     datetime64[ns]
dtype: object

In [18]:
df['Order Date'].min()

Timestamp('2014-01-03 00:00:00')

In [19]:
df['Order Date'].max()

Timestamp('2017-12-30 00:00:00')

In [20]:
df['Ship Date'].min()

Timestamp('2014-01-07 00:00:00')

In [21]:
df['Ship Date'].max()

Timestamp('2018-01-05 00:00:00')

In [22]:
df['Shipping Days'] = (df['Ship Date'] - df['Order Date']).dt.days

In [23]:
df['Shipping Days'].describe()

count    9994.000000
mean        3.958175
std         1.747567
min         0.000000
25%         3.000000
50%         4.000000
75%         5.000000
max         7.000000
Name: Shipping Days, dtype: float64

In [24]:
df['Shipping Days'].min()

np.int64(0)

In [25]:
df['Shipping Days'].max()

np.int64(7)

In [26]:
(df['Shipping Days'] < 0).sum()

np.int64(0)

In [27]:
df['Sales'].sum()

np.float64(2297200.8603000003)

In [28]:
df['Profit'].sum()

np.float64(286397.0217)

In [29]:
df['Quantity'].sum()

np.int64(37873)

In [30]:
profit_margin = (df['Profit'].sum() / df['Sales'].sum()) * 100
profit_margin

np.float64(12.467217240315604)

In [31]:
category_summary = df.groupby('Category')[['Sales', 'Profit']].sum()
category_summary

,Sales,Profit
Category,,
Furniture,741999.7953,18451.2728
Office Supplies,719047.0320,122490.8008
Technology,836154.0330,145454.9481


In [32]:
category_summary['Profit Margin'] = (
    category_summary['Profit'] / category_summary['Sales']
) * 100

category_summary

,Sales,Profit,Profit Margin
Category,,,
Furniture,741999.7953,18451.2728,2.486695
Office Supplies,719047.0320,122490.8008,17.035158
Technology,836154.0330,145454.9481,17.395712


In [33]:
region_summary = df.groupby('Region')[['Sales', 'Profit']].sum()
region_summary

,Sales,Profit
Region,,
Central,501239.8908,39706.3625
East,678781.2400,91522.7800
South,391721.9050,46749.4303
West,725457.8245,108418.4489


In [34]:
region_summary['Profit Margin'] = (
    region_summary['Profit'] / region_summary['Sales']
) * 100

region_summary

,Sales,Profit,Profit Margin
Region,,,
Central,501239.8908,39706.3625,7.921629
East,678781.2400,91522.7800,13.483399
South,391721.9050,46749.4303,11.934342
West,725457.8245,108418.4489,14.944831


In [35]:
df['Discount'].describe()

count    9994.000000
mean        0.156203
std         0.206452
min         0.000000
25%         0.000000
50%         0.200000
75%         0.200000
max         0.800000
Name: Discount, dtype: float64

In [36]:
df['Discount'].value_counts().sort_index()

Discount
0.00    4798
0.10      94
0.15      52
0.20    3657
0.30     227
0.32      27
0.40     206
0.45      11
0.50      66
0.60     138
0.70     418
0.80     300
Name: count, dtype: int64

In [37]:
discount_summary = df.groupby('Discount')[['Sales', 'Profit']].sum()

discount_summary

,Sales,Profit
Discount,,
0.00,1.087908e+06,320987.6032
0.10,5.436935e+04,9029.1770
0.15,2.755852e+04,1418.9915
0.20,7.645944e+05,90337.3060
0.30,1.032267e+05,-10369.2774
0.32,1.449346e+04,-2391.1377
0.40,1.164178e+05,-23057.0504
0.45,5.484974e+03,-2493.1111
0.50,5.891854e+04,-20506.4281


In [38]:
discount_summary['Profit Margin'] = (
    discount_summary['Profit'] / discount_summary['Sales']
) * 100

discount_summary

,Sales,Profit,Profit Margin
Discount,,,
0.00,1.087908e+06,320987.6032,29.505019
0.10,5.436935e+04,9029.1770,16.607108
0.15,2.755852e+04,1418.9915,5.149012
0.20,7.645944e+05,90337.3060,11.815063
0.30,1.032267e+05,-10369.2774,-10.045155
0.32,1.449346e+04,-2391.1377,-16.498047
0.40,1.164178e+05,-23057.0504,-19.805437
0.45,5.484974e+03,-2493.1111,-45.453472
0.50,5.891854e+04,-20506.4281,-34.804712


In [39]:
discount_category = df.groupby(['Discount', 'Category'])['Profit'].sum()

discount_category

Discount  Category       
0.00      Furniture           58133.0764
          Office Supplies    130506.1064
          Technology         132348.4204
0.10      Furniture            7111.0119
          Office Supplies      1086.0808
          Technology            832.0843
0.15      Furniture            1418.9915
0.20      Furniture            6265.9491
          Office Supplies     38038.7512
          Technology          46032.6057
0.30      Furniture          -10695.3169
          Technology            326.0395
0.32      Furniture           -2391.1377
0.40      Furniture          -16187.3968
          Technology          -6869.6536
0.45      Furniture           -2493.1111
0.50      Furniture          -12871.1990
          Technology          -7635.2291
0.60      Furniture           -5944.6552
0.70      Furniture           -3894.9394
          Office Supplies    -16601.0984
          Technology         -19579.3191
0.80      Office Supplies    -30539.0392
Name: Profit, dtype: float64

In [40]:
discount_category_count = df.groupby(
    ['Discount', 'Category']
).size()

discount_category_count

Discount  Category       
0.00      Furniture           836
          Office Supplies    3129
          Technology          833
0.10      Furniture            76
          Office Supplies      16
          Technology            2
0.15      Furniture            52
0.20      Furniture           615
          Office Supplies    2201
          Technology          841
0.30      Furniture           222
          Technology            5
0.32      Furniture            27
0.40      Furniture            75
          Technology          131
0.45      Furniture            11
0.50      Furniture            54
          Technology           12
0.60      Furniture           138
0.70      Furniture            15
          Office Supplies     380
          Technology           23
0.80      Office Supplies     300
dtype: int64

In [41]:
discount_category_summary = df.groupby(
    ['Discount', 'Category']
)[['Sales', 'Profit']].sum()

discount_category_summary

Sales       Profit
Discount Category                                 
0.00     Furniture        256025.2700   58133.0764
         Office Supplies  442150.0000  130506.1064
         Technology       389733.2000  132348.4204
0.10     Furniture         46634.2470    7111.0119
         Office Supplies    4324.1490    1086.0808
         Technology         3410.9550     832.0843
0.15     Furniture         27558.5215    1418.9915
0.20     Furniture        216631.0160    6265.9491
         Office Supplies  233049.7360   38038.7512
         Technology       314913.6160   46032.6057
0.30     Furniture         99470.3500  -10695.3169
         Technology         3756.3050     326.0395
0.32     Furniture         14493.4588   -2391.1377
0.40     Furniture         45614.4060  -16187.3968
         Technology        70803.3780   -6869.6536
0.45     Furniture          5484.9740   -2493.1111
0.50     Furniture         20983.4700  -12871.1990
         Technology        37935.0700   -7635.2291
0.60     Furniture          6644.7000   -5944.6552
0.70     Furniture          2459.3820   -3894.9394
         Office Supplies   22559.3910  -16601.0984
         Technology        15601.5090  -19579.3191
0.80     Office Supplies   16963.7560  -30539.0392

In [42]:
discount_category_summary['Profit Margin'] = (
    discount_category_summary['Profit']
    / discount_category_summary['Sales']
) * 100

discount_category_summary

Sales       Profit  Profit Margin
Discount Category                                                
0.00     Furniture        256025.2700   58133.0764      22.705992
         Office Supplies  442150.0000  130506.1064      29.516252
         Technology       389733.2000  132348.4204      33.958724
0.10     Furniture         46634.2470    7111.0119      15.248476
         Office Supplies    4324.1490    1086.0808      25.116637
         Technology         3410.9550     832.0843      24.394467
0.15     Furniture         27558.5215    1418.9915       5.149012
0.20     Furniture        216631.0160    6265.9491       2.892452
         Office Supplies  233049.7360   38038.7512      16.322160
         Technology       314913.6160   46032.6057      14.617534
0.30     Furniture         99470.3500  -10695.3169     -10.752266
         Technology         3756.3050     326.0395       8.679793
0.32     Furniture         14493.4588   -2391.1377     -16.498047
0.40     Furniture         45614.4060  -16187.3968     -35.487466
         Technology        70803.3780   -6869.6536      -9.702438
0.45     Furniture          5484.9740   -2493.1111     -45.453472
0.50     Furniture         20983.4700  -12871.1990     -61.339707
         Technology        37935.0700   -7635.2291     -20.127099
0.60     Furniture          6644.7000   -5944.6552     -89.464614
0.70     Furniture          2459.3820   -3894.9394    -158.370656
         Office Supplies   22559.3910  -16601.0984     -73.588416
         Technology        15601.5090  -19579.3191    -125.496316
0.80     Office Supplies   16963.7560  -30539.0392    -180.025221

In [43]:
discount_subcategory_count = df.groupby(
    ['Discount', 'Sub-Category']
).size()

discount_subcategory_count

Discount  Sub-Category
0.00      Accessories     471
          Appliances      271
          Art             498
          Binders         337
          Bookcases        60
          Chairs          133
          Copiers          22
          Envelopes       152
          Fasteners       128
          Furnishings     571
          Labels          239
          Machines         29
          Paper           857
          Phones          311
          Storage         530
          Supplies        117
          Tables           72
0.10      Appliances       16
          Chairs           76
          Machines          2
0.15      Bookcases        52
0.20      Accessories     304
          Appliances      112
          Art             298
          Binders         573
          Bookcases        46
          Chairs          250
          Copiers          37
          Envelopes       102
          Fasteners        89
          Furnishings     248
          Labels          125
          Machine

In [44]:
discount_subcategory_summary = df.groupby(
    ['Discount', 'Sub-Category']
)[['Sales', 'Profit']].sum()

discount_subcategory_summary

Sales      Profit
Discount Sub-Category                         
0.00     Accessories   118370.3100  35289.2539
         Appliances     78066.1900  23183.7361
         Art            18014.6400   5380.6006
         Binders        81829.4800  39314.4507
         Bookcases      31935.9800   6075.7117
         Chairs         91060.7300  21933.0961
         Copiers        76449.1800  35556.1349
         Envelopes      10606.4500   4976.9844
         Fasteners       1822.8400    652.2052
         Furnishings    61449.8000  16847.9689
         Labels          9239.5600   4422.0960
         Machines       71034.0000  27137.8223
         Paper          53157.8700  25329.4677
         Phones        123879.7100  34365.2093
         Storage       157853.7600  25528.1715
         Supplies       31559.2100   1718.3942
         Tables         71578.7600  13276.2997
0.10     Appliances      4324.1490   1086.0808
         Chairs         46634.2470   7111.0119
         Machines        3410.9550    832.0843
0.15     Bookcases      27558.5215   1418.9915
0.20     Accessories    49010.0080   6647.3818
         Appliances     21759.2880   2497.8297
         Art             9104.1520   1147.1864
         Binders        85442.6400  29417.8090
         Bookcases      26841.4960    130.5022
         Chairs        120748.6240   4283.1750
         Copiers        56159.0480  17878.7148
         Envelopes       5869.9520   1987.1923
         Fasteners       1201.4400    297.3130
         Furnishings    23610.6640   2155.8299
         Labels          3246.7520   1124.1580
         Machines       37954.5680   4970.2024
         Paper          25321.3360   8724.1016
         Phones        171789.9920  16536.3067
         Storage        65989.8480  -4249.3451
         Supplies       15114.3280  -2907.4937
         Tables         45430.2320   -303.5580
0.30     Bookcases       4282.6980   -555.8726
         Chairs         70005.5020  -6737.1167
         Machines        3756.3050    326.0395
         Tables         25182.1500  -3402.3276
0.32     Bookcases      14493.4588  -2391.1377
0.40     Copiers        16919.8020   2182.9752
         Machines       19546.2240  -2666.8434
         Phones         34337.3520  -6385.7854
         Tables         45614.4060 -16187.3968
0.45     Tables          5484.9740  -2493.1111
0.50     Bookcases       7308.4600  -4255.8117
         Machines       37935.0700  -7635.2291
         Tables         13675.0100  -8615.3873
0.60     Furnishings     6644.7000  -5944.6552
0.70     Binders        22559.3910 -16601.0984
         Bookcases       2459.3820  -3894.9394
         Machines       15601.5090 -19579.3191
0.80     Appliances      3382.5340  -8629.6412
         Binders        13581.2220 -21909.3980

In [45]:
discount_subcategory_summary['Profit Margin'] = (
    discount_subcategory_summary['Profit']
    / discount_subcategory_summary['Sales']
) * 100

discount_subcategory_summary

Sales      Profit  Profit Margin
Discount Sub-Category                                        
0.00     Accessories   118370.3100  35289.2539      29.812589
         Appliances     78066.1900  23183.7361      29.697538
         Art            18014.6400   5380.6006      29.867933
         Binders        81829.4800  39314.4507      48.044361
         Bookcases      31935.9800   6075.7117      19.024660
         Chairs         91060.7300  21933.0961      24.086229
         Copiers        76449.1800  35556.1349      46.509505
         Envelopes      10606.4500   4976.9844      46.924130
         Fasteners       1822.8400    652.2052      35.779619
         Furnishings    61449.8000  16847.9689      27.417451
         Labels          9239.5600   4422.0960      47.860461
         Machines       71034.0000  27137.8223      38.203990
         Paper          53157.8700  25329.4677      47.649516
         Phones        123879.7100  34365.2093      27.740789
         Storage       157853.7600  25528.1715      16.172039
         Supplies       31559.2100   1718.3942       5.444985
         Tables         71578.7600  13276.2997      18.547820
0.10     Appliances      4324.1490   1086.0808      25.116637
         Chairs         46634.2470   7111.0119      15.248476
         Machines        3410.9550    832.0843      24.394467
0.15     Bookcases      27558.5215   1418.9915       5.149012
0.20     Accessories    49010.0080   6647.3818      13.563315
         Appliances     21759.2880   2497.8297      11.479372
         Art             9104.1520   1147.1864      12.600695
         Binders        85442.6400  29417.8090      34.429892
         Bookcases      26841.4960    130.5022       0.486196
         Chairs        120748.6240   4283.1750       3.547183
         Copiers        56159.0480  17878.7148      31.835858
         Envelopes       5869.9520   1987.1923      33.853638
         Fasteners       1201.4400    297.3130      24.746388
         Furnishings    23610.6640   2155.8299       9.130747
         Labels          3246.7520   1124.1580      34.624080
         Machines       37954.5680   4970.2024      13.095136
         Paper          25321.3360   8724.1016      34.453560
         Phones        171789.9920  16536.3067       9.625885
         Storage        65989.8480  -4249.3451      -6.439392
         Supplies       15114.3280  -2907.4937     -19.236672
         Tables         45430.2320   -303.5580      -0.668185
0.30     Bookcases       4282.6980   -555.8726     -12.979496
         Chairs         70005.5020  -6737.1167      -9.623696
         Machines        3756.3050    326.0395       8.679793
         Tables         25182.1500  -3402.3276     -13.510870
0.32     Bookcases      14493.4588  -2391.1377     -16.498047
0.40     Copiers        16919.8020   2182.9752      12.901896
         Machines       19546.2240  -2666.8434     -13.643778
         Phones         34337.3520  -6385.7854     -18.597198
         Tables         45614.4060 -16187.3968     -35.487466
0.45     Tables          5484.9740  -2493.1111     -45.453472
0.50     Bookcases       7308.4600  -4255.8117     -58.231306
         Machines       37935.0700  -7635.2291     -20.127099
         Tables         13675.0100  -8615.3873     -63.000958
0.60     Furnishings     6644.7000  -5944.6552     -89.464614
0.70     Binders        22559.3910 -16601.0984     -73.588416
         Bookcases       2459.3820  -3894.9394    -158.370656
         Machines       15601.5090 -19579.3191    -125.496316
0.80     Appliances      3382.5340  -8629.6412    -255.123561
         Binders        13581.2220 -21909.3980    -161.321257

In [46]:
discount_subcategory_detail = df.groupby(
    ['Discount', 'Sub-Category']
).agg(
    Rows=('Profit', 'size'),
    Sales=('Sales', 'sum'),
    Profit=('Profit', 'sum')
)

discount_subcategory_detail

Rows        Sales      Profit
Discount Sub-Category                               
0.00     Accessories    471  118370.3100  35289.2539
         Appliances     271   78066.1900  23183.7361
         Art            498   18014.6400   5380.6006
         Binders        337   81829.4800  39314.4507
         Bookcases       60   31935.9800   6075.7117
         Chairs         133   91060.7300  21933.0961
         Copiers         22   76449.1800  35556.1349
         Envelopes      152   10606.4500   4976.9844
         Fasteners      128    1822.8400    652.2052
         Furnishings    571   61449.8000  16847.9689
         Labels         239    9239.5600   4422.0960
         Machines        29   71034.0000  27137.8223
         Paper          857   53157.8700  25329.4677
         Phones         311  123879.7100  34365.2093
         Storage        530  157853.7600  25528.1715
         Supplies       117   31559.2100   1718.3942
         Tables          72   71578.7600  13276.2997
0.10     Appliances      16    4324.1490   1086.0808
         Chairs          76   46634.2470   7111.0119
         Machines         2    3410.9550    832.0843
0.15     Bookcases       52   27558.5215   1418.9915
0.20     Accessories    304   49010.0080   6647.3818
         Appliances     112   21759.2880   2497.8297
         Art            298    9104.1520   1147.1864
         Binders        573   85442.6400  29417.8090
         Bookcases       46   26841.4960    130.5022
         Chairs         250  120748.6240   4283.1750
         Copiers         37   56159.0480  17878.7148
         Envelopes      102    5869.9520   1987.1923
         Fasteners       89    1201.4400    297.3130
         Furnishings    248   23610.6640   2155.8299
         Labels         125    3246.7520   1124.1580
         Machines        31   37954.5680   4970.2024
         Paper          513   25321.3360   8724.1016
         Phones         469  171789.9920  16536.3067
         Storage        316   65989.8480  -4249.3451
         Supplies        73   15114.3280  -2907.4937
         Tables          71   45430.2320   -303.5580
0.30     Bookcases       10    4282.6980   -555.8726
         Chairs         158   70005.5020  -6737.1167
         Machines         5    3756.3050    326.0395
         Tables          54   25182.1500  -3402.3276
0.32     Bookcases       27   14493.4588  -2391.1377
0.40     Copiers          9   16919.8020   2182.9752
         Machines        13   19546.2240  -2666.8434
         Phones         109   34337.3520  -6385.7854
         Tables          75   45614.4060 -16187.3968
0.45     Tables          11    5484.9740  -2493.1111
0.50     Bookcases       18    7308.4600  -4255.8117
         Machines        12   37935.0700  -7635.2291
         Tables          36   13675.0100  -8615.3873
0.60     Furnishings    138    6644.7000  -5944.6552
0.70     Binders        380   22559.3910 -16601.0984
         Bookcases       15    2459.3820  -3894.9394
         Machines        23   15601.5090 -19579.3191
0.80     Appliances      67    3382.5340  -8629.6412
         Binders        233   13581.2220 -21909.3980

In [47]:
discount_subcategory_detail['Profit Margin'] = (
    discount_subcategory_detail['Profit']
    / discount_subcategory_detail['Sales']
) * 100

In [48]:
discount_subcategory_detail

Rows        Sales      Profit  Profit Margin
Discount Sub-Category                                              
0.00     Accessories    471  118370.3100  35289.2539      29.812589
         Appliances     271   78066.1900  23183.7361      29.697538
         Art            498   18014.6400   5380.6006      29.867933
         Binders        337   81829.4800  39314.4507      48.044361
         Bookcases       60   31935.9800   6075.7117      19.024660
         Chairs         133   91060.7300  21933.0961      24.086229
         Copiers         22   76449.1800  35556.1349      46.509505
         Envelopes      152   10606.4500   4976.9844      46.924130
         Fasteners      128    1822.8400    652.2052      35.779619
         Furnishings    571   61449.8000  16847.9689      27.417451
         Labels         239    9239.5600   4422.0960      47.860461
         Machines        29   71034.0000  27137.8223      38.203990
         Paper          857   53157.8700  25329.4677      47.649516
         Phones         311  123879.7100  34365.2093      27.740789
         Storage        530  157853.7600  25528.1715      16.172039
         Supplies       117   31559.2100   1718.3942       5.444985
         Tables          72   71578.7600  13276.2997      18.547820
0.10     Appliances      16    4324.1490   1086.0808      25.116637
         Chairs          76   46634.2470   7111.0119      15.248476
         Machines         2    3410.9550    832.0843      24.394467
0.15     Bookcases       52   27558.5215   1418.9915       5.149012
0.20     Accessories    304   49010.0080   6647.3818      13.563315
         Appliances     112   21759.2880   2497.8297      11.479372
         Art            298    9104.1520   1147.1864      12.600695
         Binders        573   85442.6400  29417.8090      34.429892
         Bookcases       46   26841.4960    130.5022       0.486196
         Chairs         250  120748.6240   4283.1750       3.547183
         Copiers         37   56159.0480  17878.7148      31.835858
         Envelopes      102    5869.9520   1987.1923      33.853638
         Fasteners       89    1201.4400    297.3130      24.746388
         Furnishings    248   23610.6640   2155.8299       9.130747
         Labels         125    3246.7520   1124.1580      34.624080
         Machines        31   37954.5680   4970.2024      13.095136
         Paper          513   25321.3360   8724.1016      34.453560
         Phones         469  171789.9920  16536.3067       9.625885
         Storage        316   65989.8480  -4249.3451      -6.439392
         Supplies        73   15114.3280  -2907.4937     -19.236672
         Tables          71   45430.2320   -303.5580      -0.668185
0.30     Bookcases       10    4282.6980   -555.8726     -12.979496
         Chairs         158   70005.5020  -6737.1167      -9.623696
         Machines         5    3756.3050    326.0395       8.679793
         Tables          54   25182.1500  -3402.3276     -13.510870
0.32     Bookcases       27   14493.4588  -2391.1377     -16.498047
0.40     Copiers          9   16919.8020   2182.9752      12.901896
         Machines        13   19546.2240  -2666.8434     -13.643778
         Phones         109   34337.3520  -6385.7854     -18.597198
         Tables          75   45614.4060 -16187.3968     -35.487466
0.45     Tables          11    5484.9740  -2493.1111     -45.453472
0.50     Bookcases       18    7308.4600  -4255.8117     -58.231306
         Machines        12   37935.0700  -7635.2291     -20.127099
         Tables          36   13675.0100  -8615.3873     -63.000958
0.60     Furnishings    138    6644.7000  -5944.6552     -89.464614
0.70     Binders        380   22559.3910 -16601.0984     -73.588416
         Bookcases       15    2459.3820  -3894.9394    -158.370656
         Machines        23   15601.5090 -19579.3191    -125.496316
0.80     Appliances      67    3382.5340  -8629.6412    -255.123561
         Binders        233   13581.2220 -21909.3980    -161.321257

In [49]:
discount_subcategory_detail.sort_values(
    'Sales',
    ascending=False
).head(10)

Rows       Sales      Profit  Profit Margin
Discount Sub-Category                                             
0.2      Phones         469  171789.992  16536.3067       9.625885
0.0      Storage        530  157853.760  25528.1715      16.172039
         Phones         311  123879.710  34365.2093      27.740789
0.2      Chairs         250  120748.624   4283.1750       3.547183
0.0      Accessories    471  118370.310  35289.2539      29.812589
         Chairs         133   91060.730  21933.0961      24.086229
0.2      Binders        573   85442.640  29417.8090      34.429892
0.0      Binders        337   81829.480  39314.4507      48.044361
         Appliances     271   78066.190  23183.7361      29.697538
         Copiers         22   76449.180  35556.1349      46.509505

In [50]:
discount_subcategory_detail.sort_values(
    'Profit',
    ascending=False
).head(10)

Rows      Sales      Profit  Profit Margin
Discount Sub-Category                                            
0.0      Binders        337   81829.48  39314.4507      48.044361
         Copiers         22   76449.18  35556.1349      46.509505
         Accessories    471  118370.31  35289.2539      29.812589
         Phones         311  123879.71  34365.2093      27.740789
0.2      Binders        573   85442.64  29417.8090      34.429892
0.0      Machines        29   71034.00  27137.8223      38.203990
         Storage        530  157853.76  25528.1715      16.172039
         Paper          857   53157.87  25329.4677      47.649516
         Appliances     271   78066.19  23183.7361      29.697538
         Chairs         133   91060.73  21933.0961      24.086229

In [51]:
discount_subcategory_detail[
    discount_subcategory_detail['Profit'] < 0
].sort_values(
    'Profit'
)

Rows       Sales      Profit  Profit Margin
Discount Sub-Category                                             
0.80     Binders        233  13581.2220 -21909.3980    -161.321257
0.70     Machines        23  15601.5090 -19579.3191    -125.496316
         Binders        380  22559.3910 -16601.0984     -73.588416
0.40     Tables          75  45614.4060 -16187.3968     -35.487466
0.80     Appliances      67   3382.5340  -8629.6412    -255.123561
0.50     Tables          36  13675.0100  -8615.3873     -63.000958
         Machines        12  37935.0700  -7635.2291     -20.127099
0.30     Chairs         158  70005.5020  -6737.1167      -9.623696
0.40     Phones         109  34337.3520  -6385.7854     -18.597198
0.60     Furnishings    138   6644.7000  -5944.6552     -89.464614
0.50     Bookcases       18   7308.4600  -4255.8117     -58.231306
0.20     Storage        316  65989.8480  -4249.3451      -6.439392
0.70     Bookcases       15   2459.3820  -3894.9394    -158.370656
0.30     Tables          54  25182.1500  -3402.3276     -13.510870
0.20     Supplies        73  15114.3280  -2907.4937     -19.236672
0.40     Machines        13  19546.2240  -2666.8434     -13.643778
0.45     Tables          11   5484.9740  -2493.1111     -45.453472
0.32     Bookcases       27  14493.4588  -2391.1377     -16.498047
0.30     Bookcases       10   4282.6980   -555.8726     -12.979496
0.20     Tables          71  45430.2320   -303.5580      -0.668185

In [52]:
loss_making_groups = discount_subcategory_detail[
    discount_subcategory_detail['Profit'] < 0
]

In [53]:
loss_making_groups['Profit'].sum()

np.float64(-145345.4675)

In [54]:
loss_making_groups.shape[0]

20

In [55]:
loss_making_groups['Sales'].sum()

np.float64(468628.4508)

In [56]:
loss_making_margin = (
    loss_making_groups['Profit'].sum()
    / loss_making_groups['Sales'].sum()
) * 100

In [57]:
loss_making_margin

np.float64(-31.015075429560323)

In [58]:
loss_making_sales_share = (
    loss_making_groups['Sales'].sum()
    / df['Sales'].sum()
) * 100

In [59]:
loss_making_sales_share

np.float64(20.399977159106584)

In [60]:
loss_to_total_profit_ratio = (
    abs(loss_making_groups['Profit'].sum())
    / df['Profit'].sum()
) * 100

In [61]:
loss_to_total_profit_ratio

np.float64(50.74964349742748)

In [62]:
product_summary = df.groupby('Product Name').agg(
    Rows=('Profit', 'size'),
    Sales=('Sales', 'sum'),
    Profit=('Profit', 'sum')
)

In [63]:
product_summary

,Rows,Sales,Profit
Product Name,,,
"""While you Were Out"" Message Book, One Form per Page",3,25.228,10.3880
"#10 Gummed Flap White Envelopes, 100/Box",4,41.300,16.7678
#10 Self-Seal White Envelopes,4,108.682,52.1230
"#10 White Business Envelopes,4 1/8 x 9 1/2",7,488.904,223.1408
"#10- 4 1/8"" x 9 1/2"" Recycled Envelopes",10,286.672,115.2806
...,...,...,...
"iKross Bluetooth Portable Keyboard + Cell Phone Stand Holder + Brush for Apple iPhone 5S 5C 5, 4S 4",5,477.660,115.6440
iOttie HLCRIO102 Car Mount,5,215.892,-11.9940
iOttie XL Car Mount,2,223.888,-50.3748


In [64]:
product_summary.sort_values(
    'Profit'
).head(10)

,Rows,Sales,Profit
Product Name,,,
Cubify CubeX 3D Printer Double Head Print,3,11099.963,-8879.9704
Lexmark MX611dhe Monochrome Laser Printer,4,16829.901,-4589.9730
Cubify CubeX 3D Printer Triple Head Print,1,7999.980,-3839.9904
Chromcraft Bull-Nose Wood Oval Conference Tables & Bases,5,9917.640,-2876.1156
Bush Advantage Collection Racetrack Conference Table,7,9544.725,-1934.3976
GBC DocuBind P400 Electric Binding System,6,17965.068,-1878.1662
Cisco TelePresence System EX90 Videoconferencing Unit,1,22638.480,-1811.0784
Martin Yale Chadless Opener Electric Letter Opener,6,16656.200,-1299.1836
Balt Solid Wood Round Tables,4,6518.754,-1201.0581


In [65]:
product_summary['Profit Margin'] = (
    product_summary['Profit'] / product_summary['Sales']
) * 100

In [66]:
product_summary

,Rows,Sales,Profit,Profit Margin
Product Name,,,,
"""While you Were Out"" Message Book, One Form per Page",3,25.228,10.3880,41.176471
"#10 Gummed Flap White Envelopes, 100/Box",4,41.300,16.7678,40.600000
#10 Self-Seal White Envelopes,4,108.682,52.1230,47.959184
"#10 White Business Envelopes,4 1/8 x 9 1/2",7,488.904,223.1408,45.641026
"#10- 4 1/8"" x 9 1/2"" Recycled Envelopes",10,286.672,115.2806,40.213415
...,...,...,...,...
"iKross Bluetooth Portable Keyboard + Cell Phone Stand Holder + Brush for Apple iPhone 5S 5C 5, 4S 4",5,477.660,115.6440,24.210526
iOttie HLCRIO102 Car Mount,5,215.892,-11.9940,-5.555556
iOttie XL Car Mount,2,223.888,-50.3748,-22.500000


In [67]:
product_summary.sort_values(
    'Profit Margin'
).head(10)

,Rows,Sales,Profit,Profit Margin
Product Name,,,,
Eureka Disposable Bags for Sanitaire Vibra Groomer I Upright Vac,1,1.624,-4.4660,-275.000000
"Bush Westfield Collection Bookcases, Dark Cherry Finish, Fully Assembled",1,90.882,-190.8522,-210.000000
Euro Pro Shark Stick Mini Vacuum,3,170.744,-325.6332,-190.714286
Okidata B401 Printer,1,179.991,-251.9874,-140.000000
Zebra GK420t Direct Thermal/Thermal Transfer Printer,1,703.710,-938.2800,-133.333333
GBC Plasticlear Binding Covers,7,68.880,-68.4208,-99.333333
"Brother MFC-9340CDW LED All-In-One Printer, Copier Scanner",1,341.991,-319.1916,-93.333333
Epson TM-T88V Direct Thermal Printer - Monochrome - Desktop,2,1212.705,-1057.2300,-87.179487
Epson Perfection V600 Photo Scanner,1,206.991,-172.4925,-83.333333


In [68]:
product_summary[
    (product_summary['Profit'] < 0) &
    (product_summary['Sales'] >= 1000)
].sort_values(
    'Profit'
).head(10)

,Rows,Sales,Profit,Profit Margin
Product Name,,,,
Cubify CubeX 3D Printer Double Head Print,3,11099.963,-8879.9704,-80.000000
Lexmark MX611dhe Monochrome Laser Printer,4,16829.901,-4589.9730,-27.272727
Cubify CubeX 3D Printer Triple Head Print,1,7999.980,-3839.9904,-48.000000
Chromcraft Bull-Nose Wood Oval Conference Tables & Bases,5,9917.640,-2876.1156,-29.000000
Bush Advantage Collection Racetrack Conference Table,7,9544.725,-1934.3976,-20.266667
GBC DocuBind P400 Electric Binding System,6,17965.068,-1878.1662,-10.454545
Cisco TelePresence System EX90 Videoconferencing Unit,1,22638.480,-1811.0784,-8.000000
Martin Yale Chadless Opener Electric Letter Opener,6,16656.200,-1299.1836,-7.800000
Balt Solid Wood Round Tables,4,6518.754,-1201.0581,-18.424658


In [69]:
loss_making_products = product_summary[
    product_summary['Profit'] < 0
]

In [70]:
loss_making_products.shape[0]

301

In [71]:
loss_making_product_share = (
    loss_making_products.shape[0]
    / product_summary.shape[0]
) * 100

In [72]:
loss_making_product_share

16.27027027027027

In [73]:
loss_making_products[['Sales', 'Profit']].sum()

Sales     555414.4531
Profit    -77068.3808
dtype: float64

In [74]:
loss_making_sales_share = (
    loss_making_products['Sales'].sum()
    / df['Sales'].sum()
) * 100

In [75]:
loss_making_sales_share

np.float64(24.177879379144336)

In [76]:
loss_making_product_margin = (
    loss_making_products['Profit'].sum()
    / loss_making_products['Sales'].sum()
) * 100

In [77]:
loss_making_product_margin

np.float64(-13.875832789343018)

In [78]:
loss_product_names = loss_making_products.index

In [79]:
loss_making_rows = df[
    df['Product Name'].isin(loss_product_names)
]

In [80]:
loss_category_summary = loss_making_rows.groupby('Category')[['Sales', 'Profit']].sum()

In [81]:
loss_category_summary

,Sales,Profit
Category,,
Furniture,302422.0301,-37061.7363
Office Supplies,151326.2440,-12653.6112
Technology,101666.1790,-27353.0333


In [82]:
loss_category_summary['Profit Margin'] = (
    loss_category_summary['Profit']
    / loss_category_summary['Sales']
) * 100

In [83]:
loss_category_summary

,Sales,Profit,Profit Margin
Category,,,
Furniture,302422.0301,-37061.7363,-12.254972
Office Supplies,151326.2440,-12653.6112,-8.361809
Technology,101666.1790,-27353.0333,-26.904752


In [84]:
loss_making_products.sort_values(
    'Profit'
).head(20)

,Rows,Sales,Profit,Profit Margin
Product Name,,,,
Cubify CubeX 3D Printer Double Head Print,3,11099.9630,-8879.9704,-80.000000
Lexmark MX611dhe Monochrome Laser Printer,4,16829.9010,-4589.9730,-27.272727
Cubify CubeX 3D Printer Triple Head Print,1,7999.9800,-3839.9904,-48.000000
Chromcraft Bull-Nose Wood Oval Conference Tables & Bases,5,9917.6400,-2876.1156,-29.000000
Bush Advantage Collection Racetrack Conference Table,7,9544.7250,-1934.3976,-20.266667
GBC DocuBind P400 Electric Binding System,6,17965.0680,-1878.1662,-10.454545
Cisco TelePresence System EX90 Videoconferencing Unit,1,22638.4800,-1811.0784,-8.000000
Martin Yale Chadless Opener Electric Letter Opener,6,16656.2000,-1299.1836,-7.800000
Balt Solid Wood Round Tables,4,6518.7540,-1201.0581,-18.424658


In [85]:
top_10_product_losses = loss_making_products.sort_values(
    'Profit'
).head(10)['Profit'].sum()

In [86]:
top_10_product_losses

np.float64(-29458.3708)

In [87]:
top_10_loss_share = (
    abs(top_10_product_losses)
    / abs(loss_making_products['Profit'].sum())
) * 100

In [88]:
top_10_loss_share

np.float64(38.22367940549751)

In [89]:
df[
    df['Product Name'] == 'Cubify CubeX 3D Printer Double Head Print'
]

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit,Shipping Days
3151,3152,CA-2015-147830,2015-12-15,2015-12-18,First Class,NF-18385,Natalie Fritzler,Consumer,United States,Newark,...,East,TEC-MA-10000418,Technology,Machines,Cubify CubeX 3D Printer Double Head Print,1799.994,2,0.7,-2639.9912,3
4218,4219,CA-2017-149881,2017-04-01,2017-04-03,First Class,NC-18535,Nick Crebassa,Corporate,United States,San Francisco,...,West,TEC-MA-10000418,Technology,Machines,Cubify CubeX 3D Printer Double Head Print,4799.984,2,0.2,359.9988,2
7772,7773,CA-2016-108196,2016-11-25,2016-12-02,Standard Class,CS-12505,Cindy Stewart,Consumer,United States,Lancaster,...,East,TEC-MA-10000418,Technology,Machines,Cubify CubeX 3D Printer Double Head Print,4499.985,5,0.7,-6599.9780,7


In [90]:
loss_making_rows.groupby('Discount')[['Sales', 'Profit']].sum()

,Sales,Profit
Discount,,
0.00,171866.5100,20789.3215
0.10,8070.2820,830.9614
0.15,16090.1090,438.8066
0.20,155116.0400,-7414.8238
0.30,39909.1070,-7166.5295
0.32,10479.7316,-2190.3675
0.40,64968.9600,-19593.8355
0.45,4875.6565,-2234.0536
0.50,52525.8350,-18124.6388


In [91]:
loss_discount_summary = loss_making_rows.groupby(
    'Discount'
)[['Sales', 'Profit']].sum()

loss_discount_summary['Profit Margin'] = (
    loss_discount_summary['Profit']
    / loss_discount_summary['Sales']
) * 100

In [92]:
loss_discount_summary

,Sales,Profit,Profit Margin
Discount,,,
0.00,171866.5100,20789.3215,12.096203
0.10,8070.2820,830.9614,10.296560
0.15,16090.1090,438.8066,2.727182
0.20,155116.0400,-7414.8238,-4.780179
0.30,39909.1070,-7166.5295,-17.957128
0.32,10479.7316,-2190.3675,-20.900989
0.40,64968.9600,-19593.8355,-30.158764
0.45,4875.6565,-2234.0536,-45.820570
0.50,52525.8350,-18124.6388,-34.506141


# Executive Summary

## Business Question

Management wants to understand why profitability varies across the business despite strong sales performance. This analysis examines overall profitability, category and regional differences, product-level losses, and the relationship between discounts and recorded profit.

## Key Findings

* The dataset contains **9,994 order-line records representing 5,009 unique orders**.
* Total recorded sales are approximately **USD 2.30 million**, with approximately **USD 286,397 in recorded profit**, giving an overall recorded profit margin of approximately **12.47%**.
* Profitability varies substantially by category. **Furniture** recorded a much lower aggregate profit margin than **Office Supplies** and **Technology**, despite generating substantial sales.
* Regional profitability also varies. The **West** region recorded the highest aggregate sales, profit, and recorded profit margin among the four regions, while the **Central** region recorded the lowest aggregate profit margin.
* At the product level, **301 of 1,850 unique products (16.27%) were loss-making overall**.
* These loss-making products generated approximately **USD 555,414 in sales** and a combined recorded loss of approximately **USD 77,068**, representing about **24.18% of total sales**.
* The ten products with the largest dollar losses accounted for approximately **38.22% of the total recorded loss among loss-making products**, indicating that losses were concentrated among a relatively small group of products.
* Discount levels show a strong descriptive association with profitability in the analyzed data. Higher discount levels generally correspond with increasingly negative recorded profit margins, particularly within products that are loss-making overall.
* However, the analysis **does not establish that discounts caused the losses**. Discount levels may be associated with product mix, sub-category, region, customer characteristics, or other factors.

## Management Investigation Areas

The results suggest several areas that management could investigate further:

1. Products generating substantial dollar losses should be reviewed individually.
2. Furniture profitability warrants investigation because of its comparatively low aggregate recorded margin.
3. High-discount transactions should be examined alongside product, customer, regional, and order-level characteristics rather than in isolation.
4. Products with very small sales volumes should be interpreted cautiously because extreme percentage margins can result from small denominators.
5. Any future analysis should preserve the dataset grain and validate joins, aggregations, and metric definitions before drawing conclusions.

## Analytical Limitation

This analysis is descriptive and based on the fields available in the Sample Superstore dataset. It identifies patterns and areas for investigation but does not establish causal relationships. In particular, the relationship between discount and profitability should not be interpreted as evidence that discounts independently caused the observed losses.

## Portfolio Note

This project was completed as an independent data-analysis exercise using the Kaggle Sample Superstore dataset. The purpose is to demonstrate data cleaning, validation, exploratory analysis, metric definition, business interpretation, and critical evaluation of analytical conclusions.

In [93]:
import sqlite3

conn = sqlite3.connect("../data/superstore.db")

df.to_sql(
    "superstore",
    conn,
    if_exists="replace",
    index=False
)

print("Superstore data loaded into SQLite successfully.")

Superstore data loaded into SQLite successfully.


In [94]:
query = """
SELECT
    SUM(Sales) AS Total_Sales,
    SUM(Profit) AS Total_Profit,
    ROUND(
        SUM(Profit) * 100.0 / SUM(Sales),
        2
    ) AS Profit_Margin_Percent
FROM superstore;
"""

pd.read_sql_query(query, conn)

,Total_Sales,Total_Profit,Profit_Margin_Percent
0,2.297201e+06,286397.0217,12.47


In [95]:
query = """
SELECT
    Category,
    SUM(Sales) AS Total_Sales,
    SUM(Profit) AS Total_Profit,
    ROUND(
        SUM(Profit) * 100.0 / SUM(Sales),
        2
    ) AS Profit_Margin_Percent
FROM superstore
GROUP BY Category
ORDER BY Profit_Margin_Percent DESC;
"""

pd.read_sql_query(query, conn)

,Category,Total_Sales,Total_Profit,Profit_Margin_Percent
0,Technology,836154.0330,145454.9481,17.40
1,Office Supplies,719047.0320,122490.8008,17.04
2,Furniture,741999.7953,18451.2728,2.49


In [96]:
query = """
SELECT
    Region,
    SUM(Sales) AS Total_Sales,
    SUM(Profit) AS Total_Profit,
    ROUND(
        SUM(Profit) * 100.0 / SUM(Sales),
        2
    ) AS Profit_Margin_Percent
FROM superstore
GROUP BY Region
ORDER BY Profit_Margin_Percent DESC;
"""

pd.read_sql_query(query, conn)

,Region,Total_Sales,Total_Profit,Profit_Margin_Percent
0,West,725457.8245,108418.4489,14.94
1,East,678781.2400,91522.7800,13.48
2,South,391721.9050,46749.4303,11.93
3,Central,501239.8908,39706.3625,7.92


In [97]:
query = """
SELECT
    Discount,
    COUNT(*) AS Order_Line_Count,
    SUM(Sales) AS Total_Sales,
    SUM(Profit) AS Total_Profit,
    ROUND(
        SUM(Profit) * 100.0 / SUM(Sales),
        2
    ) AS Profit_Margin_Percent
FROM superstore
GROUP BY Discount
ORDER BY Discount;
"""

pd.read_sql_query(query, conn)

,Discount,Order_Line_Count,Total_Sales,Total_Profit,Profit_Margin_Percent
0,0.00,4798,1.087908e+06,320987.6032,29.51
1,0.10,94,5.436935e+04,9029.1770,16.61
2,0.15,52,2.755852e+04,1418.9915,5.15
3,0.20,3657,7.645944e+05,90337.3060,11.82
4,0.30,227,1.032267e+05,-10369.2774,-10.05
5,0.32,27,1.449346e+04,-2391.1377,-16.50
6,0.40,206,1.164178e+05,-23057.0504,-19.81
7,0.45,11,5.484974e+03,-2493.1111,-45.45
8,0.50,66,5.891854e+04,-20506.4281,-34.80
9,0.60,138,6.644700e+03,-5944.6552,-89.46


In [98]:
query = """
SELECT
    "Product Name",
    COUNT(*) AS Order_Line_Count,
    SUM(Sales) AS Total_Sales,
    SUM(Profit) AS Total_Profit,
    ROUND(
        SUM(Profit) * 100.0 / SUM(Sales),
        2
    ) AS Profit_Margin_Percent
FROM superstore
GROUP BY "Product Name"
HAVING SUM(Profit) < 0
ORDER BY Total_Profit ASC;
"""

pd.read_sql_query(query, conn)

,Product Name,Order_Line_Count,Total_Sales,Total_Profit,Profit_Margin_Percent
0,Cubify CubeX 3D Printer Double Head Print,3,11099.963,-8.879970e+03,-80.00
1,Lexmark MX611dhe Monochrome Laser Printer,4,16829.901,-4.589973e+03,-27.27
2,Cubify CubeX 3D Printer Triple Head Print,1,7999.980,-3.839990e+03,-48.00
3,Chromcraft Bull-Nose Wood Oval Conference Tabl...,5,9917.640,-2.876116e+03,-29.00
4,Bush Advantage Collection Racetrack Conference...,7,9544.725,-1.934398e+03,-20.27
...,...,...,...,...,...
296,"Brites Rubber Bands, 1 1/2 oz. Box",3,13.068,-5.148000e-01,-3.94
297,Rubber Band Ball,4,58.344,-2.992000e-01,-0.51
298,"Acco PRESSTEX Data Binder with Storage Hooks, ...",4,62.946,-1.614000e-01,-0.26
299,Premier Electric Letter Opener,8,2641.608,-7.105427e-15,-0.00
